source/common/http/path_utility.cc

```diff
@@ -92,13 +82,15 @@ PathUtil::UnescapeSlashesResult PathUtil::unescapeSlashes(RequestHeaderMap& head
   }
   const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
 
-  // TODO(yanavlasov): optimize this by adding case insensitive matcher
   std::string decoded_path{path};
-  unescapeInPath(decoded_path, "%2F", "/");
-  unescapeInPath(decoded_path, "%2f", "/");
-  unescapeInPath(decoded_path, "%5C", "\\");
-  unescapeInPath(decoded_path, "%5c", "\\");
-  headers.setPath(absl::StrCat(decoded_path, query));
+  static const std::vector<std::pair<absl::string_view, absl::string_view>> replacements{
+      {"%2F", "/"},
+      {"%2f", "/"},
+      {"%5C", "\\"},
+      {"%5c", "\\"},
+  };
+  headers.setPath(absl::StrCat(absl::StrReplaceAll(decoded_path, replacements), query));
```

Comment:
One more suggestion:

```suggestion
  headers.setPath(absl::StrCat(absl::StrReplaceAll(path, replacements), query));
```

And remove the `decoded_path` variable above.

Reply:
OK